test/Fencer/Rules/Test.hs

```diff
@@ -85,13 +85,13 @@ test_rulesLoadRulesRecursively =
 -- the old one intact.
 test_rulesLimitUnitChange :: TestTree
 test_rulesLimitUnitChange =
-  withResource createServerAppState destroyServerAppState $ \ioLogIdState ->
+  withServerAppState $ \ioLogIdState ->
     testCase "A rule limit unit change on rule reloading" $ do
       Temp.withSystemTempDirectory "fencer-config-unit" $ \tempDir -> do
         createDirectoryIfMissing True (tempDir </> dir)
 
         definitions1 <- writeLoad tempDir merchantLimitsText1
-        (_, _, state) <- ioLogIdState
+        (_, _, _, state) <- ioLogIdState
```

Comment:
If `LogIdState` was a `data` with named field, then you could use a field selector here instead of increasing the tuple each time `LogIdSt` gets updated.

Reply:
True. I updated it.